server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchResponseFormat.java

```diff
@@ -320,12 +321,11 @@ private static void formatIssueComments(SearchResponseData data, Issues.Issue.Bu
         }
 
         // On a root project, parentProjectId is null but projectId is equal to itself, which make no sense.
-        if (dto.projectUuid() != null && dto.parentProjectId() != null) {
+        if (!uuid.equals(dto.getRootUuid())) {
           ComponentDto project = data.getComponentByUuid(dto.projectUuid());
           builder.setProjectId(project.getId());
-        }
-        if (dto.parentProjectId() != null) {
-          builder.setSubProjectId(dto.parentProjectId());
+          ComponentDto subProject = data.getComponentByUuid(dto.getRootUuid());
+          builder.setSubProjectId(subProject.getId());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'subProject' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)